### PR TITLE
Fork latest opensc code, add feitian java card with PK01C(PKI) applet support

### PR DESCRIPTION
**Pull request by FeitianSmartcardReader**
Add Feitian java card with PK01C applet support, more information, please check below link.

Any questions, contact me anytime, thanks

https://github.com/OpenSC/OpenSC/pull/484#issuecomment-138268574

### CODE REVIEW

**Comment by frankmorgner** [comment]
Could you please squash the commits into one leaving a single comprehensive commit message?

File: src/libopensc/cards.h [hunk 1]
```diff
@@ -161,6 +161,13 @@ enum {
 	SC_CARD_TYPE_ENTERSAFE_3K,
 	SC_CARD_TYPE_ENTERSAFE_FTCOS_PK_01C,
 	SC_CARD_TYPE_ENTERSAFE_FTCOS_EPASS2003,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_PK_01C,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_PK_01C_T0,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_H10CR_PK_01C_T1,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_D11CR_PK_01C_T1,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_C21C_PK_01C_T1,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_A22CR_PK_01C_T1,
+        SC_CARD_TYPE_ENTERSAFE_EJAVA_A40CR_PK_01C_T1,
```
Comment (frankmorgner) [inline src/libopensc/cards.h]: whitespace error, use tabs!

**Comment by FeitianSmartcardReader** [comment]
Hi May I know what's the status of this issue?

**Comment by frankmorgner** [comment]
@FeitianSmartcardReader, see my comment above... If this is solved, the changes look good to be merged.

**Comment by FeitianSmartcardReader** [comment]
Thanks for your reply.
It solved already, please check 
https://github.com/FeitianSmartcardReader/OpenSC/commit/f139139a9d7acc81e4528cb2277f054779268f6d

**Comment by frankmorgner** [comment]
Thanks for fixing the whitespaces. I think you overlooked the second remark:

> Could you please squash the commits into one leaving a single comprehensive commit message?

**Comment by FeitianSmartcardReader** [comment]
Hi Frank

The requests we had add new ATR with our PKI applet support. So I need re-submit a pull request or you can change comprehensive by you? Thanks

**Comment by frankmorgner** [comment]
For the sake of your future contributions, I am leaving this exercise to you...

**Comment by frankmorgner** [comment]
You may rewrite your git history of FeitianSmartcardReader:master, which also updates this PR. Or you may create a new PR. It's up to you.

**Comment by asyd** [comment]
:heart:

**Comment by FeitianSmartcardReader** [comment]
Got, thanks

**Comment by frankmorgner** [comment]
closed with #625

**Pull request closed without merging**